src/_pytest/config/__init__.py

```diff
@@ -1125,11 +1125,21 @@ def _mark_plugins_for_rewrite(self, hook) -> None:
             # We don't autoload from setuptools entry points, no need to continue.
             return
 
+        def _get_files(dist):
+            # dist.files does not make sense for dists
+            # who are not stored on a filesystem
+            # at least pyoxidizer does throw a
+            # not implemented assertion in this case
+            try:
+                return dist.files
```

Comment:
Hi @franzhaas, thanks for the PR! 

However, according to [the api reference](https://importlib-metadata.readthedocs.io/en/latest/api.html#importlib_metadata.Distribution.files), `dist.files` should return `List of PackagePath for this distribution or None`, so seems like pyoxidizer is wrong in raising `NotImplementedError`.

I suggest reporting/opening a PR in `pyoxidizer` to fix that instead.

Reply:
I understand what you are pointing to.

The dist is not in a file system, therefore the concept of path becomes questionable. Returning  None, should indicate that meta data is missing, and a empty list should indicate empty meta data.

I have sympathy for pyoxidizer to raise notimplentederror. At this point, i can not see an obvious value that is both, correct, and helpfull.

Is there a API to disable the assertion rewriting, from within python that i did overlook? There is already a command line argument. To have this accessible would solve my problem as well.

Reply:
> The dist is not in a file system, therefore the concept of path becomes questionable. Returning None, should indicate that meta data is missing, and a empty list should indicate empty meta data.

I understand, but that point should be raised with the `imporlib-metadata` maintainers then, to extend the interface to allow for `NotImplementedError`.

> Is there a API to disable the assertion rewriting, from within python that i did overlook? There is already a command line argument. To have this accessible would solve my problem as well.

How are you executing pytest exactly? `pytest.main()`?

Reply:
Thanks for the hint!

I use numpy.test(), which calls pytest.main...

The solution with disabeling assert rewrites has the disadvantage that _all_ assert rewrites do not happen, also those who would work without issue.

I still believe that this change makes sense.

Alternatively I would propose to presented the erroro in a more helpfull way. 

I post the working and the failing call here, hopefully it helps someone.

>>> numpy.test()
NumPy version 1.23.0.dev0+308.gf82b5dacc
NumPy relaxed strides checking option: True
NumPy CPU features:  SSE SSE2 SSE3 SSSE3* SSE41* POPCNT* SSE42* AVX* F16C* FMA3* AVX2* AVX512F? AVX512CD? AVX512_SKX? AVX512_CLX? AVX512_CNL?
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "numpy._pytesttester", line 204, in __call__
  File "_pytest.config", line 143, in main
  File "_pytest.config", line 318, in _prepareconfig
  File "pluggy._hooks", line 265, in __call__
  File "pluggy._manager", line 80, in _hookexec
  File "pluggy._callers", line 55, in _multicall
  File "_pytest.helpconfig", line 100, in pytest_cmdline_parse
  File "pluggy._result", line 60, in get_result
  File "pluggy._callers", line 39, in _multicall
  File "_pytest.config", line 1003, in pytest_cmdline_parse
  File "_pytest.config", line 1283, in parse
  File "_pytest.config", line 1167, in _preparse
  File "_pytest.config", line 1114, in _consider_importhook
  File "_pytest.config", line 1134, in _mark_plugins_for_rewrite
  File "_pytest.config", line 785, in _iter_rewritable_modules
  File "_pytest.config", line 1131, in <genexpr>
NotImplementedError
>>> numpy.test(extra_argv=["--assert=plain"]) 
NumPy version 1.23.0.dev0+308.gf82b5dacc
NumPy relaxed strides checking option: True
NumPy CPU features:  SSE SSE2 SSE3 SSSE3* SSE41* POPCNT* SSE42* AVX* F16C* FMA3* AVX2* AVX512F? AVX512CD? AVX512_SKX? AVX512_CLX? AVX512_CNL?
....

Reply:
> The solution with disabeling assert rewrites has the disadvantage that all assert rewrites do not happen, also those who would work without issue.

Indeed. Perhaps you can set the `PYTEST_DISABLE_PLUGIN_AUTOLOAD` environment variable to `1`? This won't rewrite assertions in plugins, but IIUC in your case you don't have plugin files laying around on disk anyway?

> I still believe that this change makes sense.

Sorry I have to disagree.

`Distribution.files` documentation is clear that it should return either a `list[PackagePath]` or `None`. Raising `NotImplementedError` is not part of the interface, so it should be fixed in `pyoxidizer` (I think it should return `None` to mean "this distribution doesn't have files on the file system).

Why is that important? Imagine if packages decided to implement the `files` property slightly differently, for example project `X` decides to raise `ValueError` instead. Is pytest supposed to change to accommodate for that too, in addition to `NotImplementedError`? That's why I mention that it should be fixed in `pyoxidizer`, or brought up to `importlib-metadata` maintainers to change the interface to allow for `NotImplementedError` to be raised (I'm not advocating for the latter, I think `None` is sufficient, but that would be a valid route if you feel strongly about it).

Sorry if I'm seeming difficult, but I don't think pytest should accommodate for implementations that are broken just because it is easy, as this can have a rippling effect on the ecosystem or hide bugs.

Reply:
Ok, i can understand your argument against the first sugestion. But I dont see why this is an argument against more helpful output.

Something along the idea of the latest commit of this PR (this comit just ilustrates the idea).